internal/resources/fabric/service_token/models.go

```diff
@@ -28,6 +29,9 @@ func buildCreateRequest(d *schema.ResourceData) fabricv4.ServiceToken {
 	}
 	serviceTokenRequest.SetExpirationDateTime(expirationTime)
 
+	descriptionConfig := d.Get("description").(string)
+	serviceTokenRequest.SetDescription(descriptionConfig)
```

Comment:
Only set this without doing a check if the description is required. If it's optional (which I believe it is) you should check for empty string before setting. This is a source of plan diff errors that you've been running into.